packages/api/src/controllers/access-control.ts

```diff
@@ -8,4 +14,66 @@ const app = Router();
 accessControl.use("/signing-key", signingKeyApp);
 app.use("/access-control", accessControl);
 
+accessControl.post(
+  "/gate",
+  authorizer({ anyAdmin: true }),
+  validatePost("access-control-gate-payload"),
+  async (req, res) => {
+    const playbackId = req.body.stream.replace("video+", "");
+    const content =
+      (await db.stream.getByPlaybackId(playbackId)) ||
+      (await db.asset.getByPlaybackId(playbackId));
+
+    if (!content) {
+      throw new ForbiddenError("Content not found");
+    }
+
+    if (content.deleted) {
+      throw new ForbiddenError("Stream is deleted");
```

Comment:
We should not differentiate between "not found" and "deleted", so I think the above check should just be `!content || content.deleted` and we return the same error on both.

Reply:
Good point!